code/game/machinery/computer/shuttle_computers.dm

```diff
@@ -142,7 +213,8 @@
 			selected_port = pick(shuttle.docking_ports - shuttle.current_port)
 
 		if(!allow_selecting_all && !(selected_port in shuttle.docking_ports))
-			return
+			if(!disk || (disk.destination != selected_port))
```

Comment:
Because you don't check the shuttle compatibility with the disk again here, I can use a href exploit to get around the shuttle restriction.